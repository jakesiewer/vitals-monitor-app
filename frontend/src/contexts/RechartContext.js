import React from 'react';

const RechartContext = React.createContext([{ name: 'Page A', Lat: 53.3837370, Lng: -6.5941090, HR: 85, amt: 160 }, { name: 'Page B', HR: 110, Lat: 53.3849580, Lng: -6.5974880, amt: 160 }, { name: 'Page C', HR: 100, Lat: 53.3841190, Lng: -6.6000700, amt: 160 }, { name: 'Page D', HR: 68, Lat: 53.3838070, Lng: -6.5999090, amt: 160 }, { name: 'Page E', HR: 58, Lat: 53.3838200, Lng: -6.5999130, amt: 160 }, { name: 'Page F', HR: 76, Lat: 53.3838200, Lng: -6.5999130, amt: 160 }, { name: 'Page G', HR: 82, Lat: 53.3838200, Lng: -6.5999130, amt: 160 }, { name: 'Page H', HR: 70, Lat: 53.3838200, Lng: -6.5999130, amt: 160 }, { name: 'Page I', HR: 90, Lat: 53.3838200, Lng: -6.5999130, amt: 160 }]);

export default RechartContext;