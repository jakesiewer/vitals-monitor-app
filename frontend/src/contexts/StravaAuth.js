import React from 'react';
import axios from 'axios';

export default function StravaAuth() {

  const Authenticate = async () => {
    // const config = {
    //   headers:{
    //     'Content-Type': 'application/json',
    //     'Accept': 'application/json',
    //     'Access-Control-Allow-Origin': 'http://localhost:3000',
    //     'Access-Control-Allow-Credentials': 'true'
    //   }
    // };

    // try {
    //   const response = await axios.get('http://localhost:3001/api/login', config)
    //   console.log(response.data);
    // } catch (error) {
    //   console.log(error)
    // }
  }

  return (
    // <button><a href="http://localhost:3001/api/login">Login with Strava</a></button>
  //   <button
  //     type="submit"
  //     className="h-10 px-5 text-indigo-100 bg-indigo-700 rounded-lg transition-colors duration-150 focus:shadow-outline hover:bg-indigo-800"
  //     onClick={Authenticate}
  //   >
  //     Login with Strava
  //   </button>
  // );
  <a href="http://localhost:3001/api/login">Login with Strava</a>
  );
}