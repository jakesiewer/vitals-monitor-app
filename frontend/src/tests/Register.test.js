// Register.test.js

import { render, fireEvent, screen, waitFor, prettyDOM } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import { AuthProvider } from '../contexts/AuthContext';
import Register from '../components/accounts//Register';

const customRender = (ui, options) =>
    render(ui, { wrapper: (props) => <Router><AuthProvider>{props.children}</AuthProvider></Router>, ...options });

describe('Register Component', () => {
    beforeEach(() => {
        customRender(<Register />, {});
    });

    test('matches snapshot', async () => {
        await waitFor(() => {
            const { container } = customRender(<Register />, {});
            expect(prettyDOM(container)).toMatchSnapshot();
        })
    });

    test('renders registration form with email, password, and confirm password inputs', async () => {
        await waitFor(() => {
            const emailInput = screen.getByPlaceholderText('Email address');
            const passwordInput = screen.getByPlaceholderText('Password');
            const confirmPasswordInput = screen.getByPlaceholderText('Confirm Password');

            expect(emailInput).toBeInTheDocument();
            expect(passwordInput).toBeInTheDocument();
            expect(confirmPasswordInput).toBeInTheDocument();
        })
    });

    test('renders register button', async () => {
        await waitFor(() => {
            const registerButton = screen.getByRole('button', { name: /register/i });
            expect(registerButton).toBeInTheDocument();
        })
    });

    test('renders login link', async () => {
        await waitFor(() => {
            const loginLink = screen.getByText('Already have an account? Login');
            expect(loginLink).toBeInTheDocument();
            expect(loginLink).toHaveAttribute('href', '/login');
        })
    });

    test('allows user to enter email, password, and confirm password', async () => {
        await waitFor(() => {
            const emailInput = screen.getByPlaceholderText('Email address');
            const passwordInput = screen.getByPlaceholderText('Password');
            const confirmPasswordInput = screen.getByPlaceholderText('Confirm Password');

            fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
            fireEvent.change(passwordInput, { target: { value: 'testPassword' } });
            fireEvent.change(confirmPasswordInput, { target: { value: 'testPassword' } });

            expect(emailInput.value).toBe('test@example.com');
            expect(passwordInput.value).toBe('testPassword');
            expect(confirmPasswordInput.value).toBe('testPassword');
        })
    });
});
