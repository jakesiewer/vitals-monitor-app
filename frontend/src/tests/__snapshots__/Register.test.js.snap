// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component matches snapshot 1`] = `"[36m<div />[39m"`;
